File: apps/ios/ITTRehber/Models/Event.swift
import Foundation

struct Event: Codable, Identifiable, Hashable {
    let id: UUID
    let title: String
    let description: String?
    let startsAt: Date
    let endsAt: Date?
    let kanton: String
    let venue: String?
    let address: String?
    let imageURL: String?
    let status: String
    let createdAt: Date

    enum CodingKeys: String, CodingKey {
        case id
        case title
        case description
        case startsAt = "starts_at"
        case endsAt = "ends_at"
        case kanton
        case venue
        case address
        case imageURL = "image_url"
        case status
        case createdAt = "created_at"
    }
}

struct EventListResponse: Codable {
    let items: [Event]
    let total: Int
}

struct EventSubmitInput: Codable {
    var title: String
    var description: String?
    var startsAt: Date
    var endsAt: Date?
    var kanton: String
    var venue: String?
    var address: String?
    var imageURL: String?
    var submitterEmail: String?

    enum CodingKeys: String, CodingKey {
        case title
        case description
        case startsAt = "starts_at"
        case endsAt = "ends_at"
        case kanton
        case venue
        case address
        case imageURL = "image_url"
        case submitterEmail = "submitter_email"
    }
}

File: apps/backend/app/schemas/event.py
"""Event request/response shapes."""

from __future__ import annotations

from datetime import datetime
from uuid import UUID

from pydantic import BaseModel, ConfigDict, EmailStr, Field

from app.models.event import EventStatus


class EventIn(BaseModel):
    title: str = Field(min_length=2, max_length=200)
    description: str | None = Field(default=None, max_length=4000)
    starts_at: datetime
    ends_at: datetime | None = None
    kanton: str = Field(min_length=2, max_length=2)
    venue: str | None = Field(default=None, max_length=200)
    address: str | None = Field(default=None, max_length=256)
    image_url: str | None = Field(default=None, max_length=1024)
    submitter_email: EmailStr | None = None


class EventOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: UUID
    title: str
    description: str | None
    starts_at: datetime
    ends_at: datetime | None
    kanton: str
    venue: str | None
    address: str | None
    image_url: str | None
    status: EventStatus
    created_at: datetime


class EventListOut(BaseModel):
    items: list[EventOut]
    total: int
